### PR TITLE
feat: CNS-955: add provider description and spec identity

### PR DESCRIPTION
**Pull request by oren-lava**
In this PR, I added new fields to the stake entry struct and spec struct.

For the stake entry, I added the a description field of the Staking module's type Description. This new field replaces the moniker field. Also, I implemented a migrator that populated the new description field with the old moniker. The migrator was tested on testnet successfully.

For the spec, I added an identity field.

### CODE REVIEW

File: x/pairing/types/message_stake_provider.go [hunk 1]
```diff
@@ -60,10 +62,6 @@ func (msg *MsgStakeProvider) ValidateBasic() error {
 		return sdkerrors.Wrapf(legacyerrors.ErrInvalidAddress, "invalid provider address (%s)", err)
 	}
 
-	if len(msg.Moniker) > MAX_LEN_MONIKER {
```
Comment (Yaroms) [inline x/pairing/types/message_stake_provider.go:63]: validate description

Comment (oren-lava) [reply x/pairing/types/message_stake_provider.go:63]: done. see https://github.com/lavanet/lava/commit/44fa0e05fafba5cf83e6def25c4262a30f490b08

**Review by omerlavanet — CHANGES_REQUESTED** [review]

File: x/pairing/types/message_stake_provider.go [hunk 2]
```diff
@@ -47,6 +48,7 @@ func (msg *MsgStakeProvider) GetSignBytes() []byte {
 	return sdk.MustSortJSON(bz)
 }
 
+// TODO: add validation to new fields
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/types/message_stake_provider.go]: done. see https://github.com/lavanet/lava/commit/f8e6295f437300f6f6d7317b537919606dae7a3a

File: x/pairing/keeper/unstaking.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ func (k Keeper) UnstakeEntry(ctx sdk.Context, validator, chainID, creator, unsta
 		"address":     existingEntry.GetAddress(),
 		"chainID":     existingEntry.GetChain(),
 		"geolocation": strconv.FormatInt(int64(existingEntry.GetGeolocation()), 10),
-		"moniker":     existingEntry.GetMoniker(),
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/keeper/unstaking.go:72]: done. see https://github.com/lavanet/lava/commit/f8e6295f437300f6f6d7317b537919606dae7a3a

File: x/pairing/keeper/unstaking.go [hunk 4]
```diff
@@ -135,7 +135,7 @@ func (k Keeper) UnstakeEntryForce(ctx sdk.Context, chainID, provider, unstakeDes
 				"address":     existingEntry.GetAddress(),
 				"chainID":     existingEntry.GetChain(),
 				"geolocation": strconv.FormatInt(int64(existingEntry.GetGeolocation()), 10),
-				"moniker":     existingEntry.GetMoniker(),
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/keeper/unstaking.go:138]: done. see https://github.com/lavanet/lava/commit/f8e6295f437300f6f6d7317b537919606dae7a3a

File: x/pairing/keeper/staking.go [hunk 5]
```diff
@@ -272,8 +271,6 @@ func (k Keeper) StakeNewEntry(ctx sdk.Context, validator, creator, chainID strin
 		)
 	}
 
-	details = append(details, utils.Attribute{Key: "moniker", Value: moniker})
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/keeper/staking.go:275]: done. see https://github.com/lavanet/lava/commit/f8e6295f437300f6f6d7317b537919606dae7a3a

File: x/pairing/keeper/staking.go [hunk 6]
```diff
@@ -127,14 +126,14 @@ func (k Keeper) StakeNewEntry(ctx sdk.Context, validator, creator, chainID strin
 			{Key: "provider", Value: senderAddr.String()},
 			{Key: "stakeAppliedBlock", Value: stakeAppliedBlock},
 			{Key: "stake", Value: amount},
+			utils.LogAttr("description", description.String()),
 		}
-		details = append(details, utils.Attribute{Key: "moniker", Value: moniker})
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/keeper/staking.go:131]: done. see https://github.com/lavanet/lava/commit/f8e6295f437300f6f6d7317b537919606dae7a3a

**Review by Yaroms — APPROVED** [review]

**Pull request merged**